### PR TITLE
APIGOV-24312 - Support for using additional auth when communicating to IdP

### PR DESCRIPTION
**Pull request by vivekschauhan**
- Changes to configure the agents to use client_secret_basic, client_secret_post, client_secret_jwt, private_key_jwt, tls_client_auth and self_signed_tls_client_auth
- Changes to acquire token based for the registered IdP provider on the configured auth types

### CODE REVIEW

File: pkg/authz/oauth/constants.go [hunk 1]
```diff
@@ -16,8 +16,14 @@ const (
 
 // IDP Auth type string const
 const (
-	IDPAuthTypeAccessToken = "accessToken"
-	IDPAuthTypeClient      = "client"
+	IDPAuthTypeAccessToken             = "accessToken"
+	IDPAuthTypeClient                  = "client"
+	IDPAuthTypeClientSecretBasic       = "client_secret_basic"
```
Comment (jcollins-axway) [inline pkg/authz/oauth/constants.go:21]: this must be the default from the IDP but camelCase vs underscore_seperation bothers me a little

File: pkg/config/externalidpconfig.go [hunk 2]
```diff
@@ -11,6 +11,12 @@ import (
 const (
 	accessToken              = "accessToken"
 	client                   = "client"
+	clientSecretBasic        = "client_secret_basic"
+	clientSecretPost         = "client_secret_post"
+	clientSecretJwt          = "client_secret_jwt"
+	privateKeyJwt            = "private_key_jwt"
+	tlsClientAuth            = "tls_client_auth"
+	selfSignedTlsClientAuth  = "self_signed_tls_client_auth"
```
Comment (jcollins-axway) [inline pkg/config/externalidpconfig.go:19]: do these constants and the other constants have to be defined separately?

**Review by jcollins-axway — DISMISSED** [review]

File: pkg/config/externalidpconfig.go (hunk 2 above)
Comment (vivekschauhan) [reply pkg/config/externalidpconfig.go:19]: These constants are based on the token authentication types coming from OAuth specification

Comment (vivekschauhan) [reply pkg/config/externalidpconfig.go:19]: Will use these and remove the ones in oauth package

**Pull request merged**